src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -228,9 +230,110 @@ public static char ToCharLower(int value)
+                // Convert into uppercase 'a'..'f' => 'A'..'F' and
+                // move hex letter 'A'..'F' into range 0..5.
+                Vector128<byte> t3 = (vec & Vector128.Create((byte)0xDF)) - Vector128.Create((byte)'A');
+                // And correct the range into 10..15.
+                // The non-hex letters bytes become greater than 0x0f.
+                Vector128<byte> t4 = AddSaturate(t3, Vector128.Create((byte)10));
+                // Convert '0'..'9' into nibbles 0..9. Non-digit bytes become
+                // greater than 0x0f. Finally choose the result: either valid nibble (0..9/10..15)
+                // or some byte greater than 0x0f.
+                Vector128<byte> nibbles = Vector128.Min(t2 - Vector128.Create((byte)0xF0), t4);
+                // Any high bit is a sign that input is not a valid hex data
+                if (!Utf16Utility.AllCharsInVector128AreAscii(vec1 | vec2) ||
+                    AddSaturate(nibbles, Vector128.Create((byte)(127 - 15))).ExtractMostSignificantBits() != 0)
+                {
+                    // Input is either non-ASCII or invalid hex data
+                    break;
+                }
+                Vector128<byte> output;
+                if (Ssse3.IsSupported)
+                {
+                    output = Ssse3.MultiplyAddAdjacent(nibbles,
+                        Vector128.Create((short)0x0110).AsSByte()).AsByte();
+                }
+                else
+                {
+                    // Workaround for missing MultiplyAddAdjacent on ARM
+                    Vector128<short> ones = Vector128.Create(0x10010).AsInt16();
+                    Vector128<short> vl = AdvSimd.Multiply(
+                        AdvSimd.ZeroExtendWideningLower(nibbles.GetLower()).AsInt16(), ones).AsInt16();
+                    Vector128<short> vu = AdvSimd.Multiply(
+                        AdvSimd.ZeroExtendWideningLower(nibbles.GetUpper()).AsInt16(), ones).AsInt16();
+                    output = AdvSimd.AddSaturate(
+                        AdvSimd.Arm64.UnzipEven(vl, vu),
+                        AdvSimd.Arm64.UnzipOdd(vl, vu)).AsByte();
+                }
+                // Accumulate output in lower INT64 half and take care about endianness
+                output = Vector128.Shuffle(output, Vector128.Create((byte)0, 2, 4, 6, 8, 10, 12, 14, 0, 0, 0, 0, 0, 0, 0, 0));
+                // Store 8 bytes in dest by given offset
+                Unsafe.WriteUnaligned(ref Unsafe.Add(ref destRef, offset / 2), output.AsUInt64().ToScalar());
+
+                offset += (nuint)Vector128<ushort>.Count * 2;
+                if (offset == (nuint)chars.Length)
+                {
+                    return true;
+                }
+                // Overlap with the current chunk for trailing elements
+                if (offset > lengthSubTwoVector128)
+                {
+                    offset = lengthSubTwoVector128;
+                }
+            } while (true);
```

Comment:
Nit
```suggestion
            } 
            while (true);
```

Reply:
@gfoidl I think the general pattern for do-while is:
```c
do
{
} while ()
```
isn't? the search in the repo seems to confirm that

Reply:
Hm, I just remember that @stephentoub did the same suggestion (while on a new line) to me in https://github.com/dotnet/runtime/pull/76337#discussion_r1015572668
Seems to be two patterns used here...

Personally I like 
```c#
do
{
}
while ();
```
more in the meantime (same as in the for-loop the "condition" is in a separate line with the body).

But I absolutely have no strong opinion here -- choose whatever is best for the repo 😉.

Reply:
neither do I, so merged 🙂 In the C++ JIT codebase our automatic code formatter prefers `} while()`